### ISSUE TITLE
CC0017 Change to auto property ignores field assignment.

### ISSUE THREAD

**Issue by thorgeirk11** [code-cracker/code-cracker#500]
Ignores field assignments.
---

The code fix for the CC0017 ignores when the field has an value set to it.

The following code:
```CSharp
private int myVar = 42;

public int MyProperty
{
    get { return myVar; }
    set { myVar = value; }
}
```

Gets fixed to the following:
```CSharp
public int MyProperty { get; set; }
```

But should be changed to:
```CSharp
public int MyProperty { get; set; } = 42;
```

---

Note that if the value is the default value this can be ignored.
So the following is totaly valid:
```CSharp
private int myVar = 0;

public int MyProperty
{
    get { return myVar; }
    set { myVar = value; }
}
// Transformed into:
public int MyProperty { get; set; }
```

----

For language level below 6 this case should be ignored by the analyser. Since setting the field is an valid way of assigning the property a default value.

**Comment by giggio** [comment]
Bug confirmed.

**Comment by thorgeirk11** [comment]
Can I tackle this ? :)

**Comment by giggio** [comment]
Sure.

**Comment by giggio** [comment]
Closed on #515
